File: sorting/card_sorting.py
import sys
import heapq
n = int(sys.stdin.readline())
cards = []
for i in range(n):
    heapq.heappush(cards,int(sys.stdin.readline()))
answer = 0
while len(cards)>1:
    card1 = heapq.heappop(cards)
    card2 = heapq.heappop(cards)
    sum = card1 + card2
    answer += sum
    heapq.heappush(cards,sum)
print(answer)

File: sorting/card_sorting.cpp
#include <iostream>
#include <queue>
using namespace std;

int main(){
    int n;
    priority_queue<int, vector<int>, greater<int>> q;
    cin >> n;
    for(int i=0; i<n; i++){
        int tmp_card;
        cin >> tmp_card;
        q.push(tmp_card);
    }
    int answer = 0;
    while(q.size() > 1){
        int card1 = q.top();
        q.pop();
        int card2 = q.top();
        q.pop();
        int sum = card1 + card2;
        answer += sum;
        q.push(sum);
    }
    cout << answer << endl;
    return 0;
}